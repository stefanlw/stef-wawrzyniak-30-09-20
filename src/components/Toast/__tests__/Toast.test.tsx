import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { Toast } from '../';

describe('Toast', () => {
  const mockCallback = jest.fn();
  const setup = () =>
    render(
      <Toast
        message="A toast message"
        ctaText="CTA test"
        callback={mockCallback}
      />,
    );

  it('should match snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should invoke callback on ctaText press', () => {
    const { getByText } = setup();
    fireEvent.press(getByText('CTA test'));
    expect(mockCallback).toBeCalledTimes(1);
  });
});
