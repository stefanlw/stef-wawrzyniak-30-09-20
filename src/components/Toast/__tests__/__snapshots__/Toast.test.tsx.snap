// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "bottom": 50,
        "height": 50,
        "paddingHorizontal": 16,
        "position": "absolute",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 6,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 8,
        },
        Object {
          "width": 718,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <Text
        style={Object {}}
      >
        A toast message
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        style={Object {}}
      >
        CTA test
      </Text>
    </View>
  </View>
</View>
`;
